kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/config/FilterApiDecl.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.javaoperatorsdk.operator.processing.GroupVersionKind;
+
+/**
+ * Information about a Kubernetes API and the corresponding filter implementation class
+ * @param group The API group
+ * @param version The API version
+ * @param kind The API kind
+ * @param className The implementation class
+ */
+public record FilterApiDecl(String group, String version, String kind, String className) {
```

Comment:
I know we don't need it today, but should we also include `imageRef` here?

Reply:
We can add that when we get to that point. Or when we're actually consuming this from a file outside of the tests. That's the point at which we start to see this operator config file as a contract rather than a hard coded thing.